src/server/db_slice.cc

```diff
@@ -1130,4 +1139,8 @@ void DbSlice::InvalidateSlotWatches(const SlotSet& slot_ids) {
   }
 }
 
+void DbSlice::SetDocDeletionCallback(DocDeletionCallback ddcb) {
+  doc_del_cb_ = move(ddcb);
```

Comment:
Since there can be only one, maybe `CHECK()` that `doc_del_db_` is empty?

Reply:
I set it after an index is created, so it can be set multiple times. Alternatively I can add some kind of flag to track if it was already set or not, but its only more complex

Reply:
Gotcha.
I must say that I'm not a huge fan of overriding the pointer, as it kind of tightly integrated the 2 classes.